src/main/c/com_micron_hse_KvsCursor.c

```diff
@@ -974,27 +1218,713 @@ Java_com_micron_hse_KvsCursor_seekRange__JLjava_nio_ByteBuffer_2IILjava_nio_Byte
+    const hse_err_t err = hse_kvs_cursor_seek_range(
+        cursor,
+        flags,
+        filter_min_data,
+        filter_min_len,
+        filter_max_data,
+        filter_max_len,
+        &found,
+        &found_len);
+
+    if (err) {
+        throw_new_hse_exception(env, err);
+        return NULL;
+    }
+
+    if (!found)
+        return NULL;
+
+    jbyteArray found_key = (*env)->NewByteArray(env, found_len);
+    (*env)->SetByteArrayRegion(env, found_key, 0, found_len, found);
+
+    /* Assert that no ArrayIndexOutOfBoundsException occurred. The byte array is
+     * constructed with a size of found_len, so the previous operation is
+     * infallible in theory.
+     */
+    assert(!(*env)->ExceptionCheck(env));
+
+    return found_key;
+}
+
+jbyteArray
+Java_com_micron_hse_KvsCursor_seekRange__JLjava_nio_ByteBuffer_2IILjava_nio_ByteBuffer_2III(
+    JNIEnv *env,
+    jobject cursor_obj,
+    jlong   cursor_handle,
+    jobject filter_min,
+    jint    filter_min_len,
+    jint    filter_min_pos,
+    jobject filter_max,
+    jint    filter_max_len,
+    jint    filter_max_pos,
+    jint    flags)
+{
+    (void)cursor_obj;
+
+    struct hse_kvs_cursor *cursor = (struct hse_kvs_cursor *)cursor_handle;
+
+    const void *filter_min_data = NULL;
+    if (filter_min) {
+        filter_min_data = (*env)->GetDirectBufferAddress(env, filter_min);
+
+        // Move the start address based on the position
+        filter_min_data = (uint8_t *)filter_min_data + filter_min_pos;
+    }
+
+    const void *filter_max_data = NULL;
+    if (filter_max) {
+        filter_max_data = (*env)->GetDirectBufferAddress(env, filter_max);
+
+        // Move the start address based on the position
+        filter_max_data = (uint8_t *)filter_max_data + filter_max_pos;
+    }
+
+    const void     *found;
+    size_t          found_len;
+    const hse_err_t err = hse_kvs_cursor_seek_range(
+        cursor,
+        flags,
+        filter_min_data,
+        filter_min_len,
+        filter_max_data,
+        filter_max_len,
+        &found,
+        &found_len);
+
+    if (err) {
+        throw_new_hse_exception(env, err);
+        return NULL;
+    }
+
+    if (!found)
+        return NULL;
+
+    jbyteArray found_key = (*env)->NewByteArray(env, found_len);
+    (*env)->SetByteArrayRegion(env, found_key, 0, found_len, found);
+
+    /* Assert that no ArrayIndexOutOfBoundsException occurred. The byte array is
+     * constructed with a size of found_len, so the previous operation is
+     * infallible in theory.
+     */
+    assert(!(*env)->ExceptionCheck(env));
+
+    return found_key;
+}
+
+jint
+Java_com_micron_hse_KvsCursor_seekRange__J_3BI_3BI_3BII(
+    JNIEnv    *env,
+    jobject    cursor_obj,
+    jlong      cursor_handle,
+    jbyteArray filter_min,
+    jint       filter_min_len,
+    jbyteArray filter_max,
+    jint       filter_max_len,
+    jbyteArray found_buf,
+    jint       found_buf_sz,
+    jint       flags)
+{
+    (void)cursor_obj;
```

Comment:
What does this do?

Reply:
Silences the unused warning from `-pedantic`